go-controller/pkg/clustermanager/pod/ip_allocator.go

```diff
@@ -34,20 +34,16 @@ type PodIPAllocator struct {
 
 	podLister listers.PodLister
 	kube      kube.Interface
-
-	// function used to update the pod, replaceable mainly for testing purposes
-	updatePodWithAllocationOrRollback func(listers.PodLister, kube.Interface, *corev1.Pod, util.AllocateToPodWithRollbackFunc) error
```

Comment:
could you explain why we're not doing the same for DPU ? 
@jcaamano

Reply:
hmm not sure I understand in the context of the code you are highlighting. The same as what?

Reply:
I mean that DPU still uses the code you said you were removing in the top-most commit: 
https://github.com/ovn-org/ovn-kubernetes/pull/3620/files#diff-2e2e03f067b90c926504d0f11b39ed85685ffc2f0f98de26d69bc08c490e9bddR222

Quoting your top-most commit msg:
> Unexport AllocatePodAnnotationWithRollback and add AllocatePodAnnotation
which takes no PodAnnotation argument and only returns a PodAnnotation
the pod has already been updated with.

Thus I assumed "everyone" would use this new form.

Reply:
After writing my msg above I noticed that `UpdatePodAllocationWithRetryOrRollback` != `AllocatePodAnnotationWithRollback` .

Reply:
Its `allocatePodAnnotationWithRollback` now, not exported so nobody of the util package can use that without some kungfu

Reply:
Changed `UpdatePodAllocationWithRetryOrRollback` to `UpdatePodWithRetryOrRollback`. No point in being so precise with the naming if it brings confusion. Let me know what you think.

Reply:
github is not picking up the latest push for some reason. We might just need to be patient.

Reply:
thank you. I do prefer that name.